pynxtools_xps/phi/spe_pro_phi.py

```diff
@@ -0,0 +1,1549 @@
+            XPS data filepath.
+
+        **kwargs : dict
+            commentprefix : str
+                Prefix for comments in xy file. The default is "#".
+            n_headerlines: int
+                number of header_lines in each data block.
+
+        Returns
+        -------
+        list
+            Flat list of dictionaries containing one spectrum each.
+
+        """
+        self.raw_data = self._read_lines(file)
+        header, data = self._separate_header_and_data()
+        # l = b'\x01'.join(data)
+
+        self.parse_header_into_metadata(header)
+        regions = self.parse_spectral_regions(header)
+        areas = self.parse_spatial_areas(header)
+
+        self.add_regions_and_areas_to_spectra(regions, areas)
+
+        self.parse_binary_header(data)
+        self.parse_data_into_spectra(data)
+
+        self.add_metadata_to_each_spectrum()
+
+        return self.spectra
+
+    def _read_lines(self, file: Union[str, Path]):
+        """
+        Read in all lines from the file as a list of strings.
+
+        Parameters
+        ----------
+        file : str
+            XPS data filepath.
+
+        Returns
+        -------
+        None.
+
+        """
+        with open(file, mode="rb") as txt_file:  # "utf-8"
+            data = txt_file.read()
+
+        return data
+
+    def _separate_header_and_data(self):
+        """
+        Separate header (with metadata) from data.
+
+        Returns
+        -------
+        None.
+
+        """
+        try:  # Windows
+            header, data = self.raw_data.split(b"EOFH\r\n")
+            header = header.decode("ascii").split("\r")
+            header = [line.strip() for line in header if line.strip()]
+
+        except ValueError:  # Linux, MacOS
+            header, data = self.raw_data.split(b"EOFH\n")
+            header = header.decode("ascii").split("\n")
+            header = [line.strip() for line in header if line.strip()]
+
+        return header, data
+
+    def parse_header_into_metadata(self, header: List[str]):
+        """
+        Parse header into PHiMetadata dataclass.
+
+        Parameters
+        ----------
+        header : str
+            Header data for one spectrum as a String.
+
+        """
+
+        def map_keys(key: str, channel_count: int):
+            """Maps key names for better identification of fields."""
+            if key.startswith("neut"):
+                key = key.replace("neut_", "neutral_")
+
+            elif key.startswith("prof"):
+                key = key.replace("prof_", "profiling_")
+
+            elif key.startswith("x_ray"):
+                key = key.replace("x_ray", "xray")
+
+            elif key.startswith("egun_neut"):
+                key = key.replace("egun_neut", "flood_gun")
+
+            elif key.startswith("sxi_lens"):
+                key += "_voltage"
+
+            elif key.startswith("channel _info"):
+                key = f"channel_{channel_count}_info"
+                channel_count += 1
```

Comment:
Consider making this a dict and looping over the dict here

Reply:
done